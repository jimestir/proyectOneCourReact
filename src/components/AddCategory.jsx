import { useState } from 'react'

function AddCategory ({ onNewCategory }) {
  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({ target }) => {
    setInputValue(target.value)
  }

  const onSubmit = (e) => {
    e.preventDefault()
    const inputValueTrim = inputValue.trim()
    if (inputValueTrim.length <= 1) return
    onNewCategory(inputValue)
    setInputValue('')
  }
  return (
    <form onSubmit={onSubmit}>
      <input placeholder='Search gifs...' onChange={onInputChange} value={inputValue} type='text' />
    </form>
  )
}

export { AddCategory }
