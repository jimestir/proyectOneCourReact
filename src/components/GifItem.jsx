import PropTypes from 'prop-types'
function GifItem ({ id, title, url }) {
  return (
    <div className='card'>
      <img src={url} alt={title} />
      <span>{id}</span>
    </div>
  )
}

export { GifItem }

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
