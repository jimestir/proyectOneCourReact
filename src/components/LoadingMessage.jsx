function LoadingMessage ({ isLoading }) {
  return (isLoading && <span>Loading...</span>)
}

export { LoadingMessage }
