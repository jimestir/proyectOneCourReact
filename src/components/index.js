export * from './LoadingMessage'
export * from './GifItem'
export * from './GifGrid'
export * from './AddCategory'
