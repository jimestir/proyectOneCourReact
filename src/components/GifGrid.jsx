import { GifItem } from '../components/GifItem'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { LoadingMessage } from './LoadingMessage'
function GifGrid ({ category }) {
  const { images, isLoading } = useFetchGifs(category)
  return (
    <>
      <h2 className='title-category'>{category}</h2>
      <LoadingMessage isLoading={isLoading} />
      <div className='card-grid'>
        {images.map(images => <GifItem key={images.id} {...images} />)}
      </div>

    </>
  )
}
export { GifGrid }
