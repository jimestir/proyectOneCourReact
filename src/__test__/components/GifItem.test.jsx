import { render, screen } from '@testing-library/react'
import { GifItem } from '../../components/GifItem'

const GIF = {
  id: 1234,
  title: 'Hello i am Rick',
  url: 'http://localhost/www.google.com'
}

describe('Test in hook useFetchGifs()', () => {
  test('should has match with the snapshot', () => {
    render(<GifItem {...GIF} />)
    expect(screen.container).toMatchSnapshot()
  })
  test('should has two props, url end title', () => {
    render(<GifItem {...GIF} />)
    // screen.debug()
    // expect(screen.getByAltText(/hello i am Ric/i).alt).toBe(GIF.title)
    // expect(screen.getByAltText(/Hello i am Rick/).src).toBe(GIF.url)
    // console.log(screen.getByRole('img').src)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(GIF.url)
    expect(alt).toBe(GIF.title)
  })
})
