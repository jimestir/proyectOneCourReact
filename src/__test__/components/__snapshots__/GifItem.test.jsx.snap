// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in hook useFetchGifs() should has match with the snapshot 1`] = `undefined`;
